### PR TITLE
Enable tox --parallel 

### PR DESCRIPTION
**Pull request by graingert**
### Contributor Checklist:

- [ ] I have updated the release notes at `docs/source/NEWS.rst`
- [x] I have updated the automated tests.
- [ ] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by graingert** [comment]
I don't expect this to be useful without a GHA update, https://github.community/t/log-group-multiplexing/130543/5

**Comment by graingert** [comment]
@adiroiban this might be nicer, what do you think? Currently we have to wait for the job to finish before seeing any output - https://github.community/t/log-group-multiplexing/130543/5 but maybe that's not so bad?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,19 +63,27 @@ jobs:
...
           pip install tox
 
-      - name: tox linters
-        if: matrix.python-version != 3.5 && matrix.python-version != 'pypy3'
-        run: tox -e py-linters
+      - name: Disable pypy3
+        if: startsWith(matrix.os, 'windows-') || startsWith(matrix.os, 'macOS-')
+        run: echo '::set-env name=TOX_SKIP_ENV::pypy3.*'
+
+      - name: build coverage for pypy3 # coverage doesn't like being built in parallel
```
Comment (graingert) [inline .github/workflows/main.yml:70]:
```suggestion
      # coverage doesn't like being built in parallel, see https://git.io/JUdlT
      - name: build coverage for pypy3
```

**Comment by graingert** [comment]
Takes about a minute longer overall, and the coverage change looks worrying

**Review by adiroiban — COMMENTED** [review]
I think tox --parallel is ok for local builds.

For automated tests, I don't know if this help. I think that the VMs are 2 cores.... and a lot of tests are CPU intensive.

And is best if we see each environment as a separate status.

The coverage config change is ok.

I would say that is best to keep non-parallel runs for GHA runs.

Thanks!

**Comment by graingert** [comment]
> For automated tests, I don't know if this help. 

Could do a run without `--parallel` I think mostly we save time on the pip installs

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -49,19 +63,27 @@ jobs:
...
+        run: |
+          pypy3 -m pip install coverage>=5.3
 
       - name: tox
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: tox -e py-test-twlatest,py-test-twtrunk,publishcov,release
+          TOX_PARALLEL_NO_SPINNER: 1
+        run:
+          tox --parallel=all
```
Comment (graingert) [inline .github/workflows/main.yml:80]:
```suggestion
          tox
```

**Comment by graingert** [comment]
@adiroiban tox without `--parallel` is quite a bit slower

**Review by adiroiban — APPROVED** [review]
Thanks for the test ;)

What I wanted to say, is that for GitHub Action we can have separate jobs for each environment.

But I am ok to run one job for each OS, and inside that job to test various Python version.

You can merge it as you think is best.

Thanks!

**Comment by graingert** [comment]
@adiroiban hmm the coverage has gone down only on `while True:` lines that a clearly covered... I think this might be a coverage combine bug

**Comment by graingert** [comment]
@adiroiban https://github.com/nedbat/coveragepy/issues/1036

**Comment by adiroiban** [comment]
Thanks. I think that this can be merged.

**Pull request merged**